@types/response-time

- [ X ] I tried using the `@types/xxxx` package and had problems.

  - Authors: @urossmolnik @TonyPythoneer 

I am able to import the package however i have a problem when using it. The following package does not have typescript support for the specific function responseTime(fn)

https://github.com/expressjs/response-time/issues/13

**Currently**

`app.use(responseTime(function (req, res, time) {`

**Expected**

`app.use(responseTime((req: Request, res: Response, time: any) => {`

However it shows me an error

> [ts]
> Argument of type '(req: Request, res: Response, time: any) => void' is not assignable to parameter of type 'ResponseTimeFunction'.
>   Types of parameters 'req' and 'request' are incompatible.
>     Type 'IncomingMessage' is not assignable to type 'Request'.
>       Property 'get' is missing in type 'IncomingMessage'.
> (alias) interface Request
> import Request
> 
> 

> src/app.ts:101:35 - error TS2345: Argument of type '(req: Request, res: Response, time: any) => void' is not assignable to parameter of type 'ResponseTimeFunction'.
>   Types of parameters 'req' and 'request' are incompatible.
>     Type 'IncomingMessage' is not assignable to type 'Request'.
>       Property 'get' is missing in type 'IncomingMessage'.
> 
> 101         this.app.use(responseTime((req: Request, res: Response, time: any) => {
>                                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>

--- a/types/response-time/index.d.ts
+++ b/types/response-time/index.d.ts
@@ -1,6 +1,7 @@
 // Type definitions for response-time 2.3.2
 // Project: https://github.com/expressjs/response-time
 // Definitions by: Uros Smolnik <https://github.com/urossmolnik>, TonyYang <https://github.com/TonyPythoneer>
+//                 Dan Manastireanu <https://github.com/danmana>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.2
 
@@ -17,6 +18,7 @@
 
 
 import http = require("http");
+import express = require("express");
 
 
 export = responseTime;
@@ -29,6 +31,8 @@ declare function responseTime(options?: responseTime.ResponseTimeOptions):
     (request: http.IncomingMessage, response: http.ServerResponse, callback: (err: any) => void) => any;
 declare function responseTime(fn: responseTime.ResponseTimeFunction):
     (request: http.IncomingMessage, response: http.ServerResponse, callback: (err: any) => void) => any;
+declare function responseTime(fn: (request: express.Request, response: express.Response, time: number) => any):
+    express.RequestHandler;
 
 
 declare namespace responseTime {

--- a/types/response-time/response-time-tests.ts
+++ b/types/response-time/response-time-tests.ts
@@ -34,8 +34,8 @@ import http = require('http')
 // response time metrics tests https://github.com/expressjs/response-time#response-time-metrics //
 //////////////////////////////////////////////////////////////////////////////////////////////////
 {
-    const app = express()
-    app.use(responseTime(function (req, res, time) {
+    const app = express();
+    app.use(responseTime((req: express.Request, res: express.Response, time: number) => {
         let num: number = time;
     }));
 }

